caikit/runtime/servicers/training_management_servicer.py

```diff
@@ -69,34 +73,44 @@ def GetTrainingStatus(self, request, context):  # pylint: disable=unused-argumen
 
     def CancelTraining(self, request, context):  # pylint: disable=unused-argument
         """Cancel a training future."""
-        training_info = TrainingInfoRequest.from_proto(request)
+        training_info_request = TrainingInfoRequest.from_proto(request)
         model_future = self._get_model_future(
-            training_info.training_id, operation="cancel"
+            training_info_request.training_id, operation="cancel"
         )
         try:
             model_future.cancel()
+            training_info = model_future.get_info()
 
             reasons = []
-            if model_future.get_info().errors:
-                reasons = [str(error) for error in model_future.get_info().errors]
+            if training_info.errors:
+                reasons = [str(error) for error in training_info.errors]
 
             return TrainingStatusResponse(
-                training_id=training_info.training_id,
-                state=model_future.get_info().status,
+                training_id=model_future.id,
+                state=training_info.status,
                 reasons=reasons,
             ).to_proto()
         except CaikitCoreException as err:
+            # In the case that we get a `NOT_FOUND`, we assume that the training was canceled.
```

Comment:
Does this mean that calling `CancelTraining` with a random sequence of characters will show up as `CANCELED` and not `NOT_FOUND`? That seems maybe worse than returning `NOT_FOUND` after cancellation? Hm, that's a tricky one.

Reply:
It doesn't, since the initial call to get the model future would fail. This try only covers the `future.cancel()` and subsequent `future.get_info()`